packages/react-native-sdk/docusaurus/docs/reactnative/06-advanced/11-native-permissions.mdx

```diff
@@ -0,0 +1,180 @@
+  // by setting the permissions to the config.
+  StreamVideoRN.setPermissions({
+    isCameraPermissionGranted:
+      results[PERMISSIONS.IOS.CAMERA] === RESULTS.GRANTED,
+    isMicPermissionGranted:
+      results[PERMISSIONS.IOS.MICROPHONE] === RESULTS.GRANTED,
+  });
+```
+
+## Sync Permissions on App State Change
+In order to sync the permissions when the app comes back to foreground, we will add a listener to the app state change event.
+This will ensure that the permissions are always in sync when the user comes back to the app from background after
+changing the permissions in the app's settings screen.
+
+```tsx
+import {useEffect, useRef} from 'react';
+import {AppState, AppStateStatus, Platform} from 'react-native';
+
+// We will be using react-native-permissions lib. to manage our permissions
+import {checkMultiple, PERMISSIONS} from 'react-native-permissions';
+
+export const useSyncPermissions = () => {
+  // Check, update and sync permissions...
+  useEffect(() => {
+    // ...after parent component mounts
+    requestAndUpdatePermissions();
+  }, []);
+  // ...after user comes back to the app from background to foreground.
+  // This is important because the user might have changed the permissions in the app's
+  // settings screen while the app was in background.
+  const appStateRef = useRef(AppState.currentState);
+  useEffect(() => {
+    const handleAppStateChange = async (nextAppState: AppStateStatus) => {
+      const prevAppState = appStateRef.current;
+      if (
+        prevAppState.match(/inactive|background/) &&
+        nextAppState === 'active'
+      ) {
+        await checkAndUpdatePermissions();
+      }
+    };
+    const subscription = AppState.addEventListener(
+      'change',
+      handleAppStateChange,
+    );
+
+    return () => {
+      AppState.removeEventListener('change', handleAppStateChange);
+    };
+  }, []);
+};
+
+const checkAndUpdatePermissions = async () => {
+  if (Platform.OS === 'ios') {
+    // Check, update and sync permissions on iOS
+    const results = await checkMultiple([
+      PERMISSIONS.IOS.CAMERA,
+      PERMISSIONS.IOS.MICROPHONE,
+    ]);
+    // Sync the permissions with the Stream Video SDK
+    iOSProcessResultsAndSetToConfig(results);
+  } else if (Platform.OS === 'android') {
+    // Check, update and sync permissions on Android
+    const results = await checkMultiple([
+      PERMISSIONS.ANDROID.CAMERA,
+      PERMISSIONS.ANDROID.RECORD_AUDIO,
+    ]);
+    // Sync the permissions with the Stream Video SDK
+    androidProcessResultsAndSetToConfig(results);
+  }
+};
+
+const requestAndUpdatePermissions = async () => {...};
+const androidProcessResultsAndSetToConfig = (results) => {...};
+const iOSProcessResultsAndSetToConfig = (results) => {...};
+```
+
+## Use the Permissions Hook
+Use the permissions hook in the parent component of the call screen.
+This will ensure that the permissions are synced before the call screen is mounted.
```

Comment:
```suggestion
## Use the Permissions Hook

Use the permissions hook in the parent component of the call screen.
This will ensure that the permissions are synced before the call screen is mounted.
```